src/Our.Umbraco.StorageProviders.AWSS3/IO/AWSS3FileSystem.cs

```diff
@@ -236,18 +256,33 @@ public bool FileExists(string path)
             var request = new GetObjectMetadataRequest
             {
                 BucketName = _bucketName,
-                Key = ResolveBucketPath(path)
+                Key = ResolveBucketPath(path),
             };
 
             try
             {
-                Execute(client => client.GetObjectMetadataAsync(request));
+                var response = Execute(client => client.GetObjectMetadataAsync(request)).Result;
                 return true;
             }
             catch (FileNotFoundException)
             {
                 return false;
             }
+            catch (AggregateException ex)
+            {
+                foreach (var innerEx in ex.InnerExceptions)
+                {
+                    if (innerEx is AmazonS3Exception amazonS3Exception)
+                    {
+                        if (amazonS3Exception.StatusCode == System.Net.HttpStatusCode.NotFound)
+                        {
+                            return false;
+                        }
+                    }
+                }
+
+                throw;
+            }
```

Comment:
And allows removing this code, since the `FileNotFoundException` will be correctly thrown in the `Execute()` method again:
```suggestion
```